example/offload-template/src/DetectorConstruction.cc

```diff
@@ -0,0 +1,58 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file example/offload-template/src/DetectorConstruction.cc
+//---------------------------------------------------------------------------//
+#include "DetectorConstruction.hh"
+
+#include <G4Box.hh>
+#include <G4LogicalVolume.hh>
+#include <G4Material.hh>
+#include <G4NistManager.hh>
+#include <G4PVPlacement.hh>
+#include <G4SDManager.hh>
+#include <G4SystemOfUnits.hh>
+#include <G4VPhysicalVolume.hh>
+
+#include "SensitiveDetector.hh"
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct empty.
+ */
+DetectorConstruction::DetectorConstruction() : G4VUserDetectorConstruction() {}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Generate example geometry.
+ */
+G4VPhysicalVolume* DetectorConstruction::Construct()
+{
+    // Construct single material world volume box
+    auto const nist = G4NistManager::Instance();
+    auto const world_material = nist->FindOrBuildMaterial("G4_Pb");
+    double const world_size = 1 * m;
+    auto world_box = new G4Box("world_box", world_size, world_size, world_size);
+    auto world_lv = new G4LogicalVolume(world_box, world_material, "world_lv");
+    auto world_pv = new G4PVPlacement(
+        nullptr, G4ThreeVector(), world_lv, "world_pv", nullptr, false, 0);
+
+    return world_pv;
+}
```

Comment:
This may be too specific as this example serves a template - I would rather add 1) a hook to pick up a default gdml with an option to set a user provided gdml or 2) a simple, but generally configurable detector setup (size, materials, etc. despite that this requires additional UI input if users want to modify the geometry setup, so less straight forward).

Reply:
I'd prefer to keep it as the simplest geometry possible, just so the code compiles and shows that Celeritas offload happens. It is a skeleton just to highlight where the Celeritas' offload code just be placed. If you depend on GDML, then you depend on G4 compiled with GDML=ON, you need an geometry file from somewhere, and everything becomes more convoluted than just `cmake; make; ./run` and see the CELER_LOG messages.